clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -764,10 +781,12 @@ invariant_checker_t::check_for_pc_discontinuity(
             TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
                     shard->file_type_) ||
             // Regular fall-through.
-            (prev_instr_trace_pc + shard->prev_instr_.instr.size == memref.instr.addr) ||
+            (prev_instr_trace_pc + shard->prev_instr_.instr.size ==
+             current_memref_addr) ||
             // String loop.
-            (prev_instr_trace_pc == memref.instr.addr &&
-             (memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH ||
+            (prev_instr_trace_pc == current_memref_addr &&
```

Comment:
There are a bunch of conditions related to signal continuity. See `pre_signal_flow_continuity`. Let's make sure all of them are reflected here.

https://github.com/DynamoRIO/dynamorio/blob/0e86f7948bd7a791c9f628e2a0932cc5c08de88b/clients/drcachesim/tools/invariant_checker.cpp#L455